### PR TITLE
[Magic] Adds missing logic to Resist States

### PR DESCRIPTION
**Pull request by Frankie-hz**
**_I affirm:_**
- [x] I understand that if I do not agree to the following points by completing the checkboxes my PR will be ignored.
- [x] I have read and understood the [Contributing Guide](https://github.com/LandSandBoat/server/blob/base/CONTRIBUTING.md) and the [Code of Conduct](https://github.com/LandSandBoat/server/blob/base/CODE_OF_CONDUCT.md).
- [ ] I have _**tested my code and the things my code has changed**_ since the last commit in the PR and will test after any later commits.

## What does this pull request do?
- Corrects logic on when level correction is supposed to apply. -- Mobs vs Player: There is no dLvl for this interaction -- Player vs Mobs: Mobs do not have a disadvantage when targeted. Only a bonus.

- Adds resist rank tier change for magic burst

- Corrects logic for final resist tier results.
If the spell is a damage spell and the player does not have subtle sorcery on then you should enter the logic for resist rate / 2. In addition when the EEM tier is 150% (-3) the resist rate will always be at least 0.5.

### CODE REVIEW

**Comment by Xaver-DaRed** [comment]
> Corrects logic for final resist tier results.
If the spell is a damage spell and the player does not have subtle sorcery on then you should enter the logic for resist rate / 2. In addition when the EEM tier is 150% (-3) the resist rate will always be at least 0.5.

-3 resist tier means you are weak to it. resist rate should definately not be 0.5 minimum

File: scripts/globals/combat/magic_hit_rate.lua [hunk 1]
```diff
@@ -271,26 +281,32 @@ xi.combat.magicHitRate.calculateResistRate = function(actor, target, skillType,
         resistRank = target:getMod(elementTable[spellElement][4])
     end
 
+    -- Handles increasing res rank if a skillchain is present
+    local _, skillchainCount = FormMagicBurst(spellElement, target)
+
+    if skillchainCount > 0 then
+        resistRank = utils.clamp(resistRank - 1, -3, 11)
+    end
```
Comment (Xaver-DaRed) [inline scripts/globals/combat/magic_hit_rate.lua:289]: I dont think this is needed. Caster already gets a magic accuracy bonus in the macc step

**Comment by Frankie-hz** [comment]
> > Corrects logic for final resist tier results.
> > If the spell is a damage spell and the player does not have subtle sorcery on then you should enter the logic for resist rate / 2. In addition when the EEM tier is 150% (-3) the resist rate will always be at least 0.5.
> 
> -3 resist tier means you are weak to it. resist rate should definately not be 0.5 minimum

If a mob has the lowest tier then you hit what you call a half resist guarantee. Since it’s the weakest tier it’s impossible to get anything lower than that. Same goes for the other direction. If a mob at tier 11 (5% eem) then you’ll never be able to land an enfeeble on it because your resist will never be 0.5 or above, even if you hit the 5% dice roll. This is only for the max/min tiers. I will link everything later when I get a chance.

**Comment by Xaver-DaRed** [comment]
> > > Corrects logic for final resist tier results.
> > > If the spell is a damage spell and the player does not have subtle sorcery on then you should enter the logic for resist rate / 2. In addition when the EEM tier is 150% (-3) the resist rate will always be at least 0.5.
> > 
> > 
> > -3 resist tier means you are weak to it. resist rate should definately not be 0.5 minimum
> 
> If a mob has the lowest tier then you hit what you call a half resist guarantee. Since it’s the weakest tier it’s impossible to get anything lower than that. Same goes for the other direction. If a mob at tier 11 (5% eem) then you’ll never be able to land an enfeeble on it because your resist will never be 0.5 or above, even if you hit the 5% dice roll. This is only for the max/min tiers. I will link everything later when I get a chance.

EDIT: Ah, I see what you are doing.

I'm not liking the general organization though. nothing that cant be solved, but lets remember this is meant to be as easily read as possible.

I'll wait for the finished product though. It was just a comment for the time being

**Comment by Frankie-hz** [comment]
> > > > Corrects logic for final resist tier results.
> > > > If the spell is a damage spell and the player does not have subtle sorcery on then you should enter the logic for resist rate / 2. In addition when the EEM tier is 150% (-3) the resist rate will always be at least 0.5.
> > > 
> > > 
> > > -3 resist tier means you are weak to it. resist rate should definately not be 0.5 minimum
> > 
> > 
> > If a mob has the lowest tier then you hit what you call a half resist guarantee. Since it’s the weakest tier it’s impossible to get anything lower than that. Same goes for the other direction. If a mob at tier 11 (5% eem) then you’ll never be able to land an enfeeble on it because your resist will never be 0.5 or above, even if you hit the 5% dice roll. This is only for the max/min tiers. I will link everything later when I get a chance.
> 
> EDIT: Ah, I see what you are doing.
> 
> I'm not liking the general organization though. nothing that cant be solved, but lets remember this is meant to be as easily read as possible.
> 
> I'll wait for the finished product though. It was just a comment for the time being

But thats why I have you! 😬😬

**Comment by Xaver-DaRed** [comment]
ty for this PR Frank. It was very helpfull

**Pull request closed without merging**